### PR TITLE
v1.1.0

### PR DESCRIPTION
**Pull request by KeidsID**
### Added

- The common use of BuildContext as a dart extension.

  - `context.theme` (Shorthand for `Theme.of(context)`)
  - `context.mediaQuery` (Shorthand for `MediaQuery.of(context)`)
  - `context.scaffoldState` (Shorthand for `Scaffold.of(context)`)
  - Check **src/utils/extensions.dart** for more.

- Basic Pages to display error status based on MDN Web Docs.
  - `ErrorPages.client` (List of client error/400-499 status codes).
  - `ErrorPages.server` (List of server error/500-599 status codes).
  - `ErrorPages.getByCode()` (Get an ErrorPage based on the provided statusCode).

### Changed

- Rename the `common` folder to `src`.

### CODE REVIEW

File: README.md
Comment (FahriLol) [inline README.md:1]: No ErrorPages feature?

File: lib/src/errors/error_page.dart [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    return Scaffold(
+      body: Center(
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            Text('$statusCode', style: context.textTheme.headline3),
+            Text(name, style: context.textTheme.headline4),
+            const Divider(),
+            Text(
+              '${message ?? 'Sorry for the inconvenience'}.',
```
Comment (FahriLol) [inline lib/src/errors/error_page.dart:46]: Maybe its better without dot at the end.

File: test/src/errors/error_page_test.dart [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        message: message,
+      ),
+    );
+    await tester.pumpWidget(widget);
+
+    expect(find.text('$statusCode'), findsOneWidget);
+
+    expect(find.text(name), findsOneWidget);
+
+    expect(find.text('$message.'), findsOneWidget);
```
Comment (FahriLol) [inline test/src/errors/error_page_test.dart:25]: This also (delete the dot)

**Review by FahriLol — CHANGES_REQUESTED** [review]
Yok bisa yok

File: lib/src/errors/error_pages.dart [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    variantAlsoNegotiates: _serverErrorPagesMap[506]!,
+    insufficientStorage: _serverErrorPagesMap[507]!,
+    loopDetected: _serverErrorPagesMap[508]!,
+    notExtended: _serverErrorPagesMap[510]!,
+    networkAuthenticationRequired: _serverErrorPagesMap[511]!,
+  );
+
+  /// Get an ErrorPage based on the provided statusCode. If the statusCode is
+  /// not found, an [RangeError] will be thrown.
+  static ErrorPage getByCode(int statusCode, {Key? key, String? message}) {
```
Comment (FahriLol) [inline lib/src/errors/error_pages.dart:75]: ErrorPages.byCode() sounds better.

**Review by FahriLol — CHANGES_REQUESTED** [review]

File: lib/src/errors/error_pages.dart (hunk 3 above)
Comment (KeidsID) [reply lib/src/errors/error_pages.dart:75]: Ok say

Comment (KeidsID) [reply lib/src/errors/error_pages.dart]: Udh Say

**Review by FahriLol — CHANGES_REQUESTED** [review]
Delete "-dev" suffix on CHANGELOG.md

**Review by FahriLol — APPROVED** [review]
Gas kan merge

**Pull request merged**